internal/core/swrenderer.rs

```diff
@@ -398,29 +473,121 @@ fn prepare_scene(
 
     let prepare_scene = renderer.into_inner();
     Scene::new(
-        prepare_scene.items,
-        prepare_scene.textures,
-        prepare_scene.rounded_rectangles,
+        prepare_scene.processor.items,
+        prepare_scene.processor.textures,
+        prepare_scene.processor.rounded_rectangles,
         dirty_region,
     )
 }
 
+trait ProcessScene {
```

Comment:
I don't think that's the right extension point.
I thin TargetPixel is also not good enough because we can't store state in it.  Maybe we can have a `TargetBuffer` trait instead, that one would implement for `&mut [RGB565]`